### PR TITLE
Master kanban view kic

### PR DESCRIPTION
**Pull request by kirti-c**
Description of the issue/feature this PR addresses:
Task ID - https://www.odoo.com/web#id=33438&view_type=form&model=project.task&action=327&menu_id=4720

PAD - https://pad.odoo.com/p/r.7fad24daf7001b3b516112982e7e4021
Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/sales_team/static/src/js/sales_team.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ KanbanRecord.include({
      * @private
      */
     _openRecord: function () {
-        if (this.modelName === 'crm.team') {
+        if (this.modelName === 'crm.team' && $(".oe_kanban_crm_salesteams_list").length != 0) {
```
Comment (adr-odoo) [inline addons/sales_team/static/src/js/sales_team.js]: I think that .oe_kanban_crm_salesteams_list is not used anymore.
We should delete this file.

File: addons/project/static/src/js/project.js [hunk 2]
```diff
@@ -18,7 +18,7 @@ KanbanRecord.include({
      * @private
      */
     _openRecord: function () {
-        if (this.modelName === 'project.project') {
+        if (this.modelName === 'project.project' && $(".o_project_kanban_boxes").length != 0) {
```
Comment (adr-odoo) [inline addons/project/static/src/js/project.js]: What if record.use_tasks.raw_value is false? We will not able to open this record...

File: addons/project/views/project_views.xml [hunk 3]
```diff
@@ -209,6 +209,37 @@
             </field>
         </record>
 
+        <record id="project_view_kanban" model="ir.ui.view">
```
Comment (adr-odoo) [inline addons/project/views/project_views.xml]: @kic-odoo Why you add this view? the kanban view for project.project already exists.

File: addons/maintenance/views/maintenance_views.xml [hunk 4]
```diff
@@ -728,6 +744,24 @@
         </field>
     </record>
 
+    <record id="maintenance_team_view_kanban" model="ir.ui.view">
```
Comment (adr-odoo) [inline addons/maintenance/views/maintenance_views.xml]: @kic-odoo : same question.. the kanvan view for maintenance.team already exists.

Comment (adr-odoo) [reply addons/maintenance/views/maintenance_views.xml]: @kic-odoo : Ok, I understood, maybe add 
`<field name="view_id" ref="maintenance_team_view_kanban"/>`
in maintenance_team_action_settings

File: addons/hr_holidays/views/hr_holidays_views.xml [hunk 5]
```diff
@@ -675,8 +697,7 @@
             <field name="type">ir.actions.act_window</field>
             <field name="res_model">hr.holidays.status</field>
             <field name="view_type">form</field>
-            <field name="view_mode">tree,form</field>
-            <field name="view_id" eval="view_holiday_status_normal_tree"/>
```
Comment (adr-odoo) [inline addons/hr_holidays/views/hr_holidays_views.xml]: We don't want to use view_holiday_status_normal_tree anymore?

File: addons/hr_recruitment/views/hr_job_views.xml [hunk 6]
```diff
@@ -116,4 +116,24 @@
             </kanban>
         </field>
     </record>
+
+    <!-- hr related job position menu action -->
+    <record model="ir.actions.act_window" id="action_hr_job">
```
Comment (adr-odoo) [inline addons/hr_recruitment/views/hr_job_views.xml]: Two kanban views for hr.job.
Wich one to use for action_hr_job_config ?

File: addons/project/static/src/js/project.js (hunk 2 above)
Comment (adr-odoo) [reply addons/project/static/src/js/project.js]: use_tasks will be removed. So, it's not a problem.
I suggest this:
if (this.modelName === 'project.project' && this.$(".o_project_kanban_boxes a")) {

File: addons/account/views/account_view.xml [hunk 7]
```diff
@@ -223,6 +223,29 @@
                 </tree>
             </field>
         </record>
+        <record id="account_journal_view_kanban" model="ir.ui.view">
```
Comment (adr-odoo) [inline addons/account/views/account_view.xml]: @kic-odoo  Is it necessary to create almost the same view as account_bank_journal_view_kanban?

File: addons/project/static/src/js/project.js (hunk 2 above)
Comment (kic-odoo) [reply addons/project/static/src/js/project.js]: Done.

File: addons/project/views/project_views.xml (hunk 3 above)
Comment (adr-odoo) [reply addons/project/views/project_views.xml]: add view_id for open_view_project_all_config

Comment (atp-odoo) [reply addons/project/views/project_views.xml]: @adr-odoo ,
project.project existing kanban displya for dashboard and 
for configuration project kanban we need limited field (different from existing kanban.) 

second i didnot get your point: why we need to use view_id for open_view_project_all_config
it's already open list view (open_view_project_all_config)..

Comment (adr-odoo) [reply addons/project/views/project_views.xml]: on "open_view_project_all_config" the view mode is set to "tree,kanban,form".
But now we have two kanban views for the same model.
Maybe we want to precise wich one we want to open. I suppose the new one.

Comment (kic-odoo) [reply addons/project/views/project_views.xml]: @adr-odoo  by default the dashboard opens from config menu, i have created kanban view for project.project for config menu and defined the view_id for dashboard in open_view_project_all (dashboard action). now the new kanban view opens from this action open_view_project_all_config(by default list view),   so its working fine view_id here is not needed.

File: addons/project/static/src/js/project.js (hunk 2 above)
Comment (kic-odoo) [reply addons/project/static/src/js/project.js]: i have changed it to this, .length is needed:                                                                                                                                  if (this.modelName === 'project.project' && this.$(".o_project_kanban_boxes a").length) {

File: addons/hr_holidays/views/hr_holidays_views.xml (hunk 5 above)
Comment (kic-odoo) [reply addons/hr_holidays/views/hr_holidays_views.xml]: view_holiday_status_normal_tree is needed.

File: addons/hr_recruitment/views/hr_job_views.xml (hunk 6 above)
Comment (kic-odoo) [reply addons/hr_recruitment/views/hr_job_views.xml]: again, the other kanban view is dashboard of hr_recruitment, so i have defined the view_id in dashboard's action (action_hr_job), the other opens from config menu (by defualt the list view). so view_id is not needed for kanban view its working fine.

File: addons/account/views/account_view.xml (hunk 7 above)
Comment (kic-odoo) [reply addons/account/views/account_view.xml]: @adr-odoo , both the menu are different and their fields are different and currently dashboard opens from both menus in config. so why don't we need two different kanban views? the fields displayed in kanban views are the fields which are currently displayed in list views.

File: addons/maintenance/views/maintenance_views.xml (hunk 4 above)
Comment (adr-odoo) [reply addons/maintenance/views/maintenance_views.xml]: @kic-odoo : What about this?

Comment (kic-odoo) [reply addons/maintenance/views/maintenance_views.xml]: @adr-odoo i have added view_ids in maintenance_team_action_settings.another kanban view is dashboard.

Comment (adr-odoo) [reply addons/maintenance/views/maintenance_views.xml]: My bad.

Comment (kic-odoo) [reply addons/maintenance/views/maintenance_views.xml]: its ok.

**Comment by adr-odoo** [comment]
Merged: https://github.com/odoo/odoo/commit/36a2bfb9cfafc1642b65b783f6a63c0dbda11dd0

**Pull request closed without merging**